webgpu.h

```diff
@@ -615,32 +621,32 @@ typedef struct WGPUStencilStateFaceDescriptor {
 } WGPUStencilStateFaceDescriptor;
 
 typedef struct WGPUSurfaceDescriptor {
-    void const * nextInChain;
+    WGPUChainedStruct const * nextInChain;
     char const * label;
 } WGPUSurfaceDescriptor;
 
 typedef struct WGPUSurfaceDescriptorFromMetalLayer {
-    void const * nextInChain;
+    WGPUChainedStruct const * nextInChain;
     WGPUSType sType;
     void * layer;
 } WGPUSurfaceDescriptorFromMetalLayer;
 
-typedef struct WGPUSurfaceDescriptorFromWindowsHwnd {
-    void const * nextInChain;
+typedef struct WGPUSurfaceDescriptorFromWindowsHWND {
+    WGPUChainedStruct const * nextInChain;
     WGPUSType sType;
     void * hinstance;
     void * hwnd;
-} WGPUSurfaceDescriptorFromWindowsHwnd;
+} WGPUSurfaceDescriptorFromWindowsHWND;
 
-typedef struct WGPUSurfaceDescriptorFromWindowsXlib {
-    void const * nextInChain;
+typedef struct WGPUSurfaceDescriptorFromXlib {
+    WGPUChainedStruct const * nextInChain;
     WGPUSType sType;
     void * display;
-    uint64_t window;
-} WGPUSurfaceDescriptorFromWindowsXlib;
+    uint32_t window;
```

Comment:
Trying to find the confirmation about this, and I don't see why this would be true.
My system doesn't have "X11.h". The "Xlib.h" uses `Window` define, presumably coming from `X.h`, where it's defined as:
```cpp
#ifndef _XSERVER64
#  ifndef _XTYPEDEF_XID
#    define _XTYPEDEF_XID
typedef unsigned long XID;
#  endif
#  ifndef _XTYPEDEF_MASK
#    define _XTYPEDEF_MASK
typedef unsigned long Mask;
#  endif
#  ifndef _XTYPEDEF_ATOM
#    define _XTYPEDEF_ATOM
typedef unsigned long Atom;             /* Also in Xdefs.h */
#  endif
typedef unsigned long VisualID;
typedef unsigned long Time;
#else
#  include <X11/Xmd.h>
#  ifndef _XTYPEDEF_XID
#    define _XTYPEDEF_XID
typedef CARD32 XID;
#  endif
#  ifndef _XTYPEDEF_MASK
#    define _XTYPEDEF_MASK
typedef CARD32 Mask;
#  endif
#  ifndef _XTYPEDEF_ATOM
#    define _XTYPEDEF_ATOM
typedef CARD32 Atom;
#  endif
typedef CARD32 VisualID;
typedef CARD32 Time;
#endif

typedef XID Window;
```

So it depends on what he heck `_XSERVER64` is, and my gut tells me that it's not defined, meaning that we have `unsigned long` here.

Reply:
Also relevant - https://docs.rs/raw-window-handle/0.3.3/raw_window_handle/unix/struct.XlibHandle.html

Reply:
`X.h`:
```c
/*
 * _XSERVER64 must ONLY be defined when compiling X server sources on
 * systems where unsigned long is not 32 bits, must NOT be used in
 * client or library code.
 */
#ifndef _XSERVER64
#  ifndef _XTYPEDEF_XID
#    define _XTYPEDEF_XID
typedef unsigned long XID;
#  endif
#  ifndef _XTYPEDEF_MASK
#    define _XTYPEDEF_MASK
typedef unsigned long Mask;
#  endif
#  ifndef _XTYPEDEF_ATOM
#    define _XTYPEDEF_ATOM
typedef unsigned long Atom;		/* Also in Xdefs.h */
#  endif
typedef unsigned long VisualID;
typedef unsigned long Time;
#else
#  include <X11/Xmd.h>
#  ifndef _XTYPEDEF_XID
#    define _XTYPEDEF_XID
typedef CARD32 XID;
#  endif
#  ifndef _XTYPEDEF_MASK
#    define _XTYPEDEF_MASK
typedef CARD32 Mask;
#  endif
#  ifndef _XTYPEDEF_ATOM
#    define _XTYPEDEF_ATOM
typedef CARD32 Atom;
#  endif
typedef CARD32 VisualID;
typedef CARD32 Time;
```

The comment about `_XSERVER64` seems to indicate we shouldn't care about it, and that `defined(_XSERVER64) == (sizeof(unsigned long) == 8)`. Either it is not defined and `XID` is 32bits or it is defined and we get to the `CARD32` case.

`Xmd.h`
```c
# if defined (_LP64) || defined(__LP64__) || \
     defined(__alpha) || defined(__alpha__) || \
     defined(__ia64__) || defined(ia64) || \
     defined(__sparc64__) || \
     defined(__s390x__) || \
     defined(__amd64__) || defined(amd64) || \
     defined(__powerpc64__)
#  if !defined(__ILP32__) /* amd64-x32 is 32bit */
#   define LONG64				/* 32/64-bit architecture */
#  endif /* !__ILP32__ */
# endif

// ...

# ifdef LONG64
typedef unsigned long CARD64;
typedef unsigned int CARD32;
# else
typedef unsigned long long CARD64;
typedef unsigned long CARD32;
# endif
typedef unsigned short CARD16;
typedef unsigned char  CARD8;

typedef CARD32		BITS32;
typedef CARD16		BITS16;
```
This seems to be a slightly non-portable way to ensure `CARD32` is a 32bit unsigned integer. Also `typedef CARD32 BITS32;` seems pretty clear as to the intent.

It seems like `raw-window-handle` should be fixed (or maybe they'll tell us we're wrong).

Reply:
I think I misread the raw-window-handle code, it says `pub window: c_ulong,`, which I believe corresponds to u32 on most systems (not `usize`). So this change here should be good :+1: